src/main/kotlin/org/elm/ide/intentions/MakeDecoderIntention.kt

```diff
@@ -0,0 +1,187 @@
+package org.elm.ide.intentions
+
+import org.elm.lang.core.imports.ImportAdder.Import
+import org.elm.lang.core.lookup.ElmLookup
+import org.elm.lang.core.psi.ElmFile
+import org.elm.lang.core.psi.elements.ElmTypeDeclaration
+import org.elm.lang.core.toElmLowerId
+import org.elm.lang.core.types.*
+import org.elm.lang.core.types.TyInProgressBinding.alias
+
+class MakeDecoderIntention : AnnotationBasedGeneratorIntention() {
+    override fun getText() = "Generate Decoder"
+
+    override fun getRootIfApplicable(annotationTy: Ty): Ty? {
+        if (annotationTy !is TyUnion) return null
+        if (annotationTy.module != "Json.Decode" || annotationTy.name != "Decoder") return null
+        return annotationTy.parameters.singleOrNull()
+    }
+
+    override fun generator(context: Context): TyFunctionGenerator {
+        return DecoderGenerator(context.file, context.ty, context.name)
+    }
+}
+
+private class DecoderGenerator(
+        file: ElmFile,
+        root: Ty,
+        private val functionName: String
+) : TyFunctionGenerator(file, root) {
+    /** true if we need to import Json.Decode.Pipeline */
+    private var usedPipeline = false
+
+    override fun generateCode(): String {
+        // run gen on the root to kick off generation
+        val genBody = gen(root)
+        val rootFunc = funcsByTy[root]
+        funcsByTy.remove(root)
+        val body = rootFunc?.body ?: "    $genBody"
+
+        return buildString {
+            append("\n$functionName =\n")
+            append(body)
+
+            for (f in funcsByTy.values) {
+                append("\n\n\n")
+                append("-- TODO: generated code\n")
```

Comment:
Is this good, or should this message be more descriptive to make sure people know that the code probably need editing?

Reply:
how about `-- TODO: double-check generated code`?